dimod/include/dimod/quadratic_model.h

```diff
@@ -1081,17 +1056,13 @@ class QuadraticModel : public QuadraticModelBase<Bias, Index> {
     QuadraticModel() : base_type(), varinfo_() {}
 
     template <class B, class T>
-    explicit QuadraticModel(const BinaryQuadraticModel<B, T>& bqm)
-            : base_type(bqm) {
-        assert(bqm.vartype() == Vartype::BINARY ||
-               bqm.vartype() == Vartype::SPIN);
+    explicit QuadraticModel(const BinaryQuadraticModel<B, T>& bqm) : base_type(bqm) {
+        assert(bqm.vartype() == Vartype::BINARY || bqm.vartype() == Vartype::SPIN);
```

Comment:
Yes, it's for testing. We do it for performance and because it's consistent with most C++ libraries I've come across. We do test it all at the Cython/Python level because Python is a bit more picky about segmentation faults and the like.